code/game/turfs/simulated/walls.dm

```diff
@@ -37,7 +37,7 @@
 	if(!materialtype)
 		materialtype = DEFAULT_WALL_MATERIAL
 	material = SSmaterials.get_material_datum(materialtype)
-	if(!isnull(rmaterialtype))
+	if(ispath(rmaterialtype, /material))
```

Comment:
This will allow non-material values like text or numbers.

Reply:
you're saying ispath(2, /material) will return true?

Reply:
Or are you saying materialtype should also have an ispath check? Sorry, the github code snippet preview is still confusing to me.

Reply:
I am saying that this check means an invalid `reinf_material` value will not runtime, it will just never try to find a datum for it and code further down the stream will expect it to be set because it isn't null. This is related to the point I made in the comments earlier that materials are validated lower in the code.

Reply:
That code should be repaired, too. In my case code down the stream is never reached because the object is destroyed or properly configured by other code later. I only cleared out errors relevant to me. But given these errors never impacted anyone anyways, it's kind of moot to argue about it.